Sources/SwiftKuery/Connection.swift

```diff
@@ -167,16 +173,29 @@ public extension Connection {
                     }
                 }
                 else {
-                    onCompletion(.error(QueryError.syntaxError("Failed to map parameters.")))
+                    runCompletionHandler(.error(QueryError.syntaxError("Failed to map parameters.")), onCompletion: onCompletion)
+                    return
                 }
             }
             execute(convertedQuery, parameters: numberedParameters, onCompletion: onCompletion)
+        } catch  QueryError.syntaxError(let error) {
+            runCompletionHandler(.error(QueryError.syntaxError(error)), onCompletion: onCompletion)
+        } catch {
+            runCompletionHandler(.error(QueryError.syntaxError("Failed to build the query.")), onCompletion: onCompletion)
         }
-        catch  QueryError.syntaxError(let error) {
-            onCompletion(.error(QueryError.syntaxError(error)))
-        }
-        catch {
-            onCompletion(.error(QueryError.syntaxError("Failed to build the query.")))
+    }
+    
```

Comment:
Add comment to describe what these functions are, where they are used and when one should be used over the other.